pennylane/ops/op_math/exp.py

```diff
@@ -271,23 +271,25 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
             )
 
         # Change base to `Sum`/`Prod`
-        if isinstance(base, (Hamiltonian, LinearCombination)):
+        if isinstance(base, Hamiltonian):
             base = qml.dot(base.coeffs, base.ops)
         elif isinstance(base, Tensor):
             base = qml.prod(*base.obs)
 
         if isinstance(base, SProd):
             return self._recursive_decomposition(base.base, base.scalar * coeff)
 
-        if self.num_steps is not None and isinstance(base, (Hamiltonian, LinearCombination, Sum)):
-            # Apply trotter decomposition
-            coeffs = (
-                base.coeffs
-                if isinstance(base, (Hamiltonian, LinearCombination))
-                else [1] * len(base)
+        if isinstance(base, LinearCombination) and len(base.coeffs) == 1:
+            return self._recursive_decomposition(base.ops[0], coeff * base.coeffs[0])
```

Comment:
This is added so that the behaviour is the same as before. The general logic for `Sum` will return `PauliRot`, but the test cases expecting rotation gates that are equivalent to the produced `PauliRot`, but we don't recognize that they are equivalent and the test starts failing.

Reply:
Will do